### PR TITLE
Improve summary timeline

### PR DESCRIPTION
**Pull request by cameronwp**
* Cleans up summary timeline alignment
* Adds hyperlinks from the summary timeline down to procedures (just click on task titles in the summary)
* Replaces unicode ☐ with actual checkboxes
* Initializes a WIP header with some high level information and PET clock - but this isn't quite useful yet
* Minor bug fixes, typos

New summary timeline all in HTML and CSS:
image

My overarching strategy for the HTML timeline is to move as much as design and layout logic possible into njk templates and CSS.

The other big change worth mentioning is the addition of `box-sizing: border-box`. It changes HTML box [layout](https://developer.mozilla.org/en-US/docs/Web/CSS/box-sizing) to something that's a bit easier to think about. It also helped remove border considerations when dealing with summary task box heights - basically makes the borders internal to the divs and you don't need to worry about their effects on neighbors.

### CODE REVIEW

File: app/writer/procedure/HtmlProcedureWriter.js [hunk 1]
```diff
@@ -9,6 +9,11 @@ const nunjucksEnvironment = new nunjucks.Environment(
 	{ autoescape: false }
 );
 
+nunjucksEnvironment.addFilter('simplify', (str) => {
+	// remove any non a-Z, 0-9 chars, lowercase
```
Comment (jamesmontalvo3) [inline app/writer/procedure/HtmlProcedureWriter.js]: I think this is for generating an ID. Would it be better to hash the title? That way it’s more likely unique. Keeping only A-Z and 0-9 is probably unique for titles, unless someone had tasks like “Install β panel” and “Install Ω panel”. I can’t think of an example of that off the top of my head but it seems probable.

File: app/writer/timeline/HtmlTimelineWriter.js [hunk 2]
```diff
@@ -10,14 +10,16 @@ const nunjucksEnvironment = new nunjucks.Environment(
 	{ autoescape: false }
 );
 
+nunjucksEnvironment.addFilter('simplify', (str) => {
+	// remove any non a-Z, 0-9 chars, lowercase
+	return str.replace(/[^a-zA-Z0-9]/g, '').toLowerCase();
```
Comment (jamesmontalvo3) [inline app/writer/timeline/HtmlTimelineWriter.js]: Duplicate “simplify” filter. Is there a clean way to add it in one place, or should we have an `./app/helpers/nunjucks-filters.js` to hold functions like this?

File: app/writer/procedure/HtmlProcedureWriter.js (hunk 1 above)
Comment (cameronwp) [reply app/writer/procedure/HtmlProcedureWriter.js]: Yep, this is used for generating IDs. IDs are generally human readable, which is why I shied away from hashing them. I was assuming that every task would have a unique name though I hadn't considered non-ASCII letters showing up in the titles. If that's something you think could happen, I'm more than happy than happy to make this more robust.

File: app/writer/timeline/HtmlTimelineWriter.js (hunk 2 above)
Comment (cameronwp) [reply app/writer/timeline/HtmlTimelineWriter.js]: Yeah, I think we could abstract this away into a helper. I'll give it a shot.

File: app/writer/procedure/HtmlProcedureWriter.js (hunk 1 above)
Comment (jamesmontalvo3) [reply app/writer/procedure/HtmlProcedureWriter.js]: Maybe do both then? E.g. convert `My Awesome Activity` to `my-awesome-activity-a8cb12f9`?

Comment (cameronwp) [reply app/writer/procedure/HtmlProcedureWriter.js]: I like it! Will do.

**Comment by jamesmontalvo3** [comment]
@cameronwp I just pushed four commits:

- c3323a2 just fixes some indenting
- 2ed1d19 takes the `HH:mm` used in `timeline.js` and makes a function so it can be applied consistently throughout. Additionally, I've made it so it shows seconds for the counting PET clock since it was confusing to me initially when I clicked the "start" button but didn't a change. The moving seconds is good feedback.
- ba03646 creates a helper function for JSON.stringify and avoiding duplicates/circulars. I'm sure this won't be the last time this is used.
- 4dd9d34 fixes case-sensitivity of `jsonHelper.js`. Window's is case-insensitive and thus this didn't pose an issue on Windows. It's an annoying issue that bites me periodically.

**Comment by cameronwp** [comment]
LGTM!

**Pull request merged**